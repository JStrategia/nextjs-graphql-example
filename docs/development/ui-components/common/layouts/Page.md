# Page

Wraps components to be used as a page. Applies correct style for a page.

## Usage

```tsx
function MyPage() {
  return (
    <Page>
      <div />
    </Page>
  );
}
```
