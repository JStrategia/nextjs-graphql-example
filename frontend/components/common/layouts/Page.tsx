import React, { memo, ReactNode } from 'react';
import styled, { css } from 'styled-components';
import Container from './Container';

type Props = {
  children: ReactNode;
};

function Page({ children }: Props) {
  return (
    <Container>
      <ContentContainer>{children}</ContentContainer>
    </Container>
  );
}

const ContentContainer = styled.div(
  ({ theme }) => css`
    margin: ${theme.spacing.xs};
  `,
);

export default memo(Page);
