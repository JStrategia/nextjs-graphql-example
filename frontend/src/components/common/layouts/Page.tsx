import React from 'react';
import styled, { css } from 'styled-components';
import Container from 'components/common/layouts/Container';
import TopBar from 'components/common/navigation/TopBar';
import { Theme } from 'styles';

type Props = {
  readonly children: React.ReactNode;
};

function Page({ children }: Props) {
  return (
    <PageContainer>
      <TopBar />
      <ContentContainer>{children}</ContentContainer>
    </PageContainer>
  );
}

const PageContainer = styled.div(
  ({ theme }: { readonly theme: Theme }) => css`
    background-color: ${theme.colors.background};
  `,
);

const ContentContainer = styled(Container)(
  ({ theme }: { readonly theme: Theme }) => css`
    padding: ${theme.spacing.s} 0;
  `,
);

export default Page;
