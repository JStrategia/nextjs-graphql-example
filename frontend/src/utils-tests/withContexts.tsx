import React from 'react';
import ThemeProvider from 'contexts/ThemeProvider';

function withContexts(children: React.ReactNode) {
  return <ThemeProvider>{children}</ThemeProvider>;
}

export default withContexts;
