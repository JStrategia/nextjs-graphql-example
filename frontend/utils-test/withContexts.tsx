import React, { ReactNode } from 'react';
import ThemeProvider from '../contexts/ThemeProvider';

function withContexts(children: ReactNode) {
  return <ThemeProvider>{children}</ThemeProvider>;
}

export default withContexts;
