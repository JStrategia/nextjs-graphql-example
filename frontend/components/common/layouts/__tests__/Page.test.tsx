import * as React from 'react';
import { create } from 'react-test-renderer';
import Page from '../Page';
import withContexts from '../../../../utils-test/withContexts';

describe('Page', () => {
  describe('Given children', () => {
    const children = <span>Children</span>;

    it('renders', () => {
      const render = create(withContexts(<Page>{children}</Page>)).toJSON();

      expect(render).toMatchSnapshot();
    });
  });
});
