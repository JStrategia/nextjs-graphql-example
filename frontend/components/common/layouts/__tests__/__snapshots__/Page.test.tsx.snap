// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Page Given children renders 1`] = `
<div
  className="sc-bczRLJ eGDUch"
>
  <div
    className="sc-gsnTZi ksunwW"
  >
    <span>
      Children
    </span>
  </div>
</div>
`;
